pkg/controller/elasticsearch/client/client_test.go

```diff
@@ -311,6 +273,8 @@ func TestClientGetNodes(t *testing.T) {
 			Request:    req,
 		}
 	})
```

Comment:
These tests seem quite irrelevant now. We're mostly testing the mock here.
I think we can trust the go-elasticsearch client and not double-test the right function calls the right url?

However, I would maybe expect some tests for `official.go` and `official_api.go`.